gateway/policy_test.go

```diff
@@ -1397,3 +1397,57 @@ func TestPerAPIPolicyUpdate(t *testing.T) {
 		},
 	}...)
 }
+
+func TestParsePoliciesFromRPC(t *testing.T) {
+
+	objectID := apidef.NewObjectId()
+	explicitID := "explicit_pol_id"
+	tcs := []struct {
+		testName      string
+		allowExplicit bool
+		policy        user.Policy
+		expectedID    string
+	}{
+		{
+			testName:      "policy with explicit ID - allow_explicit_id false",
+			allowExplicit: false,
+			policy:        user.Policy{MID: objectID, ID: explicitID},
+			expectedID:    objectID.Hex(),
+		},
+		{
+			testName:      "policy with explicit ID - allow_explicit_id true",
+			allowExplicit: true,
+			policy:        user.Policy{MID: objectID, ID: explicitID},
+			expectedID:    explicitID,
+		},
+		{
+			testName:      "policy without explicit ID - allow_explicit_id false",
+			allowExplicit: false,
+			policy:        user.Policy{MID: objectID, ID: ""},
+			expectedID:    objectID.Hex(),
+		},
+		{
+			testName:      "policy without explicit ID - allow_explicit_id true",
+			allowExplicit: true,
+			policy:        user.Policy{MID: objectID, ID: ""},
+			expectedID:    objectID.Hex(),
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.testName, func(t *testing.T) {
+
+			policyList, err := json.Marshal([]user.Policy{tc.policy})
```

Comment:
Add assert.Len against policyList with expected length (new field part of test cases above maybe)

Reply:
I don't think this is necessary since we are creating the list right here.

Reply:
Ah, correct. I registered loadpolicies in my head, in my head it was a slice/map return 😅